src/renderers/dom/client/ReactMount.js

```diff
@@ -609,12 +609,17 @@ var ReactMount = {
       var prevWrappedElement = prevComponent._currentElement;
       var prevElement = prevWrappedElement.props;
       if (shouldUpdateReactComponent(prevElement, nextElement)) {
-        return ReactMount._updateRootComponent(
+        var publicInst = prevComponent._renderedComponent.getPublicInstance();
```

Comment:
publicInstance must be get inside updatedCallback.

see tc from https://github.com/facebook/react/pull/5125/files